File: ABC/ABC135/D.nim
import math

var dp: array[10^5+1, array[13, int64]]


proc solve(): void =
    const MOD : int64 = 10^9+7
    let
        S = stdin.readLine
        N = S.len

    dp[0][0] = 1

    for i, ch in S:
        if ch == '?':
            for j in 0..9:
                for k in 0..12:
                    var m = (k * 10 + j) mod 13
                    dp[i+1][m] += dp[i][k]
                    dp[i+1][m] = dp[i+1][m] mod MOD
        else:
            for k in 0..12:
                var m = (k * 10 + ord(ch) - ord('0')) mod 13
                dp[i+1][m] += dp[i][k]
                dp[i+1][m] = dp[i+1][m] mod MOD
    echo dp[N][5] mod MOD
    return

when is_main_module:
    solve()

File: ABC/ABC135/D.py
import sys

sys.setrecursionlimit(10 ** 7)
rl = sys.stdin.readline


def solve():
    MOD = 10 ** 9 + 7
    S = rl().rstrip()
    
    N = len(S)
    dp = [[0 for _ in range(13)] for _ in range(N + 1)]
    dp[0][0] = 1
    
    for i in range(N):
        if S[i] == '?':
            for j in range(10):
                for k in range(13):
                    dp[i + 1][(k * 10 + j) % 13] += dp[i][k] % MOD
        else:
            num = int(S[i])
            for j in range(13):
                dp[i + 1][(j * 10 + num) % 13] += dp[i][j] % MOD
    print(dp[N][5] % MOD)


if __name__ == '__main__':
    solve()
